addons/crm/models/crm_lead.py

```diff
@@ -595,8 +595,15 @@ def write(self, vals):
         if any(field in ['active', 'stage_id'] for field in vals):
             self._handle_won_lost(vals)
 
+        if len(self) == 1:
+            from_stage = self.stage_id.id
+            from_active = self.active
         write_result = super(Lead, self).write(vals)
 
+        # Recompute the probabilities if the stage or activation has changed (after the table update in _handle_won_lost)
+        if len(self) == 1 and (from_stage != self.stage_id.id or from_active != self.active):
+            self._compute_probabilities()
```

Comment:
this could be done not only if self == 1
you could store from_stage and from_active in a dict using ids as key.
After the write, recompute for the ones that changed.